### PR TITLE
fix(consensus): fix consensus height logging

### PR DESCRIPTION
**Pull request by abi87**
In https://github.com/berachain/beacon-kit/pull/2147 I added some logging for the consensus height, but values were not properly initialized. This PR fixes it.

## Summary by CodeRabbit

- **New Features**
	- Simplified state transition and verification processes by removing the `ConsensusBlockHeight` parameter, enhancing efficiency.
  
- **Bug Fixes**
	- Improved logging during payload processing by adjusting how consensus height is reported.

- **Refactor**
	- Removed the `GetConsensusBlockHeight` method from various interfaces and structs, streamlining the codebase.

- **Documentation**
	- Retained comments in methods for clarity on logic and future optimizations.

### CODE REVIEW

File: mod/consensus/pkg/cometbft/service/middleware/abci.go [hunk 1]
```diff
@@ -216,11 +216,18 @@
 	}
 
 	// notify that the beacon block has been received.
+	if req.GetHeight() < 0 {
+		return &cmtabci.ProcessProposalResponse{
+			Status: cmtabci.PROCESS_PROPOSAL_STATUS_UNKNOWN,
+		}, ErrUnexpectedNegativeHeight(nil)
+	}
```
[the remark this answers is not in the document]
Comment (abi87) [reply mod/consensus/pkg/cometbft/service/middleware/abci.go:223]: no need

File: mod/consensus/pkg/cometbft/service/middleware/errors.go [hunk 2]
```diff
@@ -33,6 +33,12 @@ var (
 		)
 	}
 
+	ErrUnexpectedNegativeHeight = func(errTimeout error) error {
+		return errors.Wrapf(errTimeout,
+			"Received consensus block with negative height",
+		)
+	}
```
[the remark this answers is not in the document]
Comment (abi87) [reply mod/consensus/pkg/cometbft/service/middleware/errors.go:40]: no need

File: mod/consensus/pkg/types/slot_data.go [hunk 3]
```diff
@@ -46,6 +46,7 @@ func (b *SlotData[AttestationDataT, SlashingInfoT]) New(
 	slashingInfo []SlashingInfoT,
 	proposerAddress []byte,
 	nextPayloadTimestamp time.Time,
+	consensusBlkHeight math.U64,
```
[the remark this answers is not in the document]
Comment (abi87) [reply mod/consensus/pkg/types/slot_data.go:49]: no need

File: mod/consensus/pkg/types/consensus_block.go [hunk 4]
```diff
@@ -38,12 +38,14 @@ func (b *ConsensusBlock[BeaconBlockT]) New(
 	beaconBlock BeaconBlockT,
 	proposerAddress []byte,
 	nextPayloadTimestamp time.Time,
+	consensusBlkHeight math.U64,
```
Comment (calbera) [inline mod/consensus/pkg/types/consensus_block.go:41]: consensusBlkHeight == slot (blk.GetSlot())

**Review by calbera — APPROVED** [review]
lgtm

**Pull request merged**